crates/sui-framework/packages/bridge/sources/committee.move

```diff
@@ -106,7 +116,40 @@ module bridge::committee {
         assert!(threshold >= required_threshold, ESignatureBelowThreshold);
     }
 
+    // This function applys the blocklist to the committee members, we won't need to run this very often so this is not gas optimised.
+    // TODO: add tests for this function
+    public(friend) fun execute_blocklist(self: &mut BridgeCommittee, blocklist: Blocklist) {
+        let blocklisted = message::blocklist_type(&blocklist) != 1;
+        let eth_addresses = message::blocklist_validator_addresses(&blocklist);
+        let list_len = vector::length(eth_addresses);
+        let list_idx = 0;
+        let member_idx = 0;
+        let pub_keys = vector::empty<vector<u8>>();
+        while (list_idx < list_len) {
+            let target_address = vector::borrow(eth_addresses, list_idx);
+            let found = false;
+            while (member_idx < vec_map::size(&self.members)) {
+                let (pub_key, member) = vec_map::get_entry_by_idx_mut(&mut self.members, member_idx);
+                let eth_address = crypto::ecdsa_pub_key_to_eth_address(*pub_key);
+                if (*target_address == eth_address) {
+                    member.blocklisted = blocklisted;
```

Comment:
This is more of a meta question: When freezing, trying to freeze if it's already frozen is an error. But when blocking, blocking when a committee member is already blocked is not an error. I was just wondering if there was a reason for this difference, or if not what your thoughts are on making them consistent.

Reply:
personally I prefer fail fast, if the value is not expected it should fail, however not sure if this will make ops more difficult? love to hear more opinions, @longbowlu @Bridgerz @dariorussi @manolisliolios